pytools/graph.py

```diff
@@ -236,6 +237,74 @@ def compute_topological_order(graph):
 
     return list(reversed(reverse_order))
 
+
+def compute_topological_order_v2(graph, key=None):
+    """Compute a toplogical order of nodes in a directed graph.
+
+    :arg graph: A :class:`collections.abc.Mapping` representing a directed
+        graph. The dictionary contains one key representing each node in the
+        graph, and this key maps to a :class:`collections.abc.Iterable` of its
+        successor nodes.
+
+    :arg key: A custom key function may be supplied to determine the order in
+        break-even cases.
+
+    :returns: A :class:`list` representing a valid topological ordering of the
+        nodes in the directed graph.
+
+    .. note::
+
+        * Has stricter requirements on the provided *graph* than
```

Comment:
Was forced to introduce `compute_topological_order_v2` because of this added requirement.

The hashability requirement of graph's keys sprung up because we are now implementing Kahn's algorithm, which was in-turn required to implement the prioritized topological sort order. The earlier implementation was broken and I was left with the conclusion that extending the DFS of `compute_topological_order` to prioritized topological sort cannot be done without hurting the time complexity.

Reply:
As discussed during meeting today, I don't think what's required of the graph keys is actually stronger than before, which means that we can have this replace `compute_topological_order`.